docs/community/meetings/2024-05-09.md

```diff
@@ -0,0 +1,201 @@
+
+- The post processing in `4_post_processing` was interleaved between some of the other steps (it mostly contains plotting and visualisation code, but also some other stuff);
+
+- The structure also had to differ between running the code on Monash University's [MASSIVE](https://research.monash.edu/en/equipments/massive) HPC platform, and on the University of Melbourne's [Spartan](https://dashboard.hpc.unimelb.edu.au/) HPC platform.
+
+**Question:** is there an automated pipeline?
+
+T.K. replied that the user had to run the code in each of the numbered folders in the correct (ascending) order, and that they wanted to make improvements for automating the dependent jobs on Spartan.
+
+> Eamon Conway: if you do ever automate it, we should share it with people (e.g., [this community](../README.md)) because people may be able to learn from it when they want to use Spartan.
+> I know how to use slurm for job management and can help you automate it.
+
+## Data visualisation
+
+!!! tip "Key message"
+
+    Producing good figures takes a lot of time, thought, and experimentation, and also a lot of coding.
+
+It was extremely hard to decide what to show, and how to show it, in order to highlight key messages.
+
+It was very easy to overwhelm the viewer with complicated plots and massive legends.
+For example, the scenarios involved three epidemic waves, and how can you show relationships between each pair of waves?
+It is relatively **simple to build** a 3D plot that shows these relationships, but the viewer can't really interpret the results.
+
+## Other activities
+
+!!! tip "Key message"
+
+    Following better practices would have required maybe 50% more time, but there wasn't funding — who will pay for this?
+
+Dedicating time to other activities was not feasible — no one had time, these projects had fixed deadlines and it was challenging to complete the work within these deadlines.
+
+As explained above, data visualisation took longer than expected.
+And sometimes the code simply wouldn't run on high-performance computing platforms.
+For example, sometimes Matlab just wouldn't load, there were intermittent failures for no apparent reason and with no useful error messages.
+
+Activities that would have been nice to do, but were not undertaken, included:
+
+- Testing (T.K. did not have time);
+- Code review (nobody else had time); and
+- Documentation — some was added afterwards, to accompany papers;
+
+> Rob Moss: we're very unlikely to get funding to explicitly cover these activities.
+> If possible, we need to allocate sufficient time in our budgets, as best as possible.
+> Practising these skills on smaller scales can also help us to use them with less overhead in larger projects.
+
+## Version control and publishing
+
+!!! tip "Key message"
+
+    This can be challenging even with all of the tools and infrastructure in place.
+
+**Question:** were all of the projects wrapped up into one central things?
```

Comment:
one central things????

Reply:
It took me a few seconds to realise what the problem was ("things" -> "thing")! Done.